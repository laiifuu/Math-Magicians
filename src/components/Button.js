import React from "react";

class Button extends React.Component {
    constructor(props) {
      super(props);
      this.state = {};
    }
  
    render(){
      return (<button className={this.props.className}>{this.props.value}</button>);
    }
}

export default Button;