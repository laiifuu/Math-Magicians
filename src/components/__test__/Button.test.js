import renderer from 'react-test-renderer';
import Button from '../Button';

const returnState = jest.fn();

describe('Testing Button component', () => {
  it('renders Button component correctly', () => {
    const tree = renderer
      .create(<Button className="width_25 grey center-text" id="AC" value="AC" onClick={returnState} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
