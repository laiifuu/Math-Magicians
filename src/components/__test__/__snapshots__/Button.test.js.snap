// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component renders Button component correctly 1`] = `
<button
  className="width_25 grey center-text"
  onClick={[MockFunction]}
  type="button"
>
  AC
</button>
`;
